includes/api/class-wc-admin-rest-admin-notes-controller.php

```diff
@@ -0,0 +1,304 @@
+	 * @var string
+	 */
+	protected $rest_base = 'admin/notes';
+
+	/**
+	 * Register the routes for admin notes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace, '/' . $this->rest_base, array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace, '/' . $this->rest_base . '/(?P<id>[\d-]+)', array(
+				'args'   => array(
+					'id' => array(
+						'description' => __( 'Unique ID for the resource.', 'wc-admin' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Get a single note.
+	 *
+	 * @param WP_REST_Request $request Request data.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function get_item( $request ) {
+		$note = WC_Admin_Notes::get_note( $request->get_param( 'id' ) );
+		if ( is_wp_error( $note ) ) {
+			return $note;
+		}
+
+		$data = $note->get_data();
+		$data = $this->prepare_item_for_response( $data, $request );
+		$data = $this->prepare_response_for_collection( $data );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Get all notes.
+	 *
+	 * @param WP_REST_Request $request Request data.
+	 * @return WP_REST_Response
+	 */
+	public function get_items( $request ) {
+		$per_page = isset( $request['per_page'] ) ? intval( $request['per_page'] ) : 10;
+		if ( $per_page <= 0 ) {
+			$per_page = 1;
+		}
+
+		$page = isset( $request['page'] ) ? intval( $request['page'] ) : 1;
+		if ( $page <= 0 ) {
+			$page = 1;
+		}
+
+		$args = array(
+			'per_page' => $per_page,
+			'page'     => $page,
+		);
+
+		$notes = WC_Admin_Notes::get_notes( 'edit', $args );
+
+		$data = array();
+		foreach ( (array) $notes as $note_obj ) {
+			$note   = $this->prepare_item_for_response( $note_obj, $request );
+			$note   = $this->prepare_response_for_collection( $note );
+			$data[] = $note;
+		}
+
+		$response = rest_ensure_response( $data );
+		$response->header( 'X-WP-Total', WC_Admin_Notes::get_notes_count() );
+
+		return $response;
+	}
+
+	/**
+	 * Check whether a given request has permission to read a single note.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function get_item_permissions_check( $request ) {
+		if ( ! wc_rest_check_manager_permissions( 'system_status', 'read' ) ) {
+			return new WP_Error( 'woocommerce_rest_cannot_view', __( 'Sorry, you cannot list resources.', 'wc-admin' ), array( 'status' => rest_authorization_required_code() ) );
```

Comment:
Since this is the item permission check, should the message be something more like "you cannot access this resource?"

Reply:
This seems to be the messaging used on other endpoints.  It also appears that this happens centrally on other endpoints.  Is there a reason not to use that function here:  https://github.com/woocommerce/wc-admin/blob/892d986d2e9ab13a07ea0f163b07cf02efb436a7/includes/api/class-wc-admin-rest-reports-controller.php#L55

Reply:
Yup, like @belcherj said, this is consistent with our other endpoints.

To answer your question, @belcherj - I have this method because the specific "type" (e.g. `system_status`) to check permissions for can vary from controller to controller.